src/python/pants/backend/python/goals/run_python_source.py

```diff
@@ -52,11 +57,29 @@ async def create_python_source_debug_adapter_request(
     field_set: PythonSourceFieldSet,
     debugpy: DebugPy,
     debug_adapter: DebugAdapterSubsystem,
+    pex_env: PexEnvironment,
+    python_setup: PythonSetup,
 ) -> RunDebugAdapterRequest:
-    run_request = await Get(RunRequest, PythonSourceFieldSet, field_set)
+    debugpy_pex = await Get(
+        Pex,
```

Comment:
After all this it seems like `Pex` and not `VenvPex` here deserves a comment why this is OK. It's implied by line 77 if you know how `PEX_PATH` combines with a `--venv` PEX, but IIUC, most of the troublems in all this was exactly not understanding all that.